<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8">
    <title>Taco</title>

    <meta name="author" content="Matthew Iyer">
    <meta name="author" content="Yashar Niknafs">
    <meta name="author" content="Balaji Pandian">

    <!-- Bootstrap CSS -->
    <link rel="stylesheet" href="css/bootstrap.min.css">
    <link rel="stylesheet" href="css/bootstrap-theme.min.css">

    <!-- Custom CSS -->
    <style>
    body {
        padding-top: 70px;
        font-family: sans-serif;
    }

    .jumbotron {
        position: relative;
        padding: 70px 10px 10px 10px;
        opacity: 1;
        text-align: center;
    }

    .jumbotron:before {
        content: '';
        display: block;
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        opacity: 0.2;
    }

    hr {
        border:none;
        border-radius:3px;
        height:5px;
        margin:auto;
        background-color:#FFCB05;
    }
    </style>

    <script src="js/jquery-2.2.1.min.js"></script>
    <script src="js/bootstrap.min.js"></script>

    <script>
      function clear_nav_bar_select() {
          var all_sections_in_navbar = $("#active_navbar li");
          all_sections_in_navbar.each(function(index) {
              $(this).attr('class', '');
          });
      }

      function clear_all_pages() {
          $("#overview").hide();
          $("#about").hide();
          $("#gettingstarted").hide();
          $("#code").hide();
          $("#faq").hide();
          $("#manual").hide();
          $("#license").hide();
      }

      function change_page_to(target) {
        clear_nav_bar_select();
        $("#active_navbar li a[href='#" + target + "']").parent().attr('class', 'active');
        clear_all_pages();
        $("#" + target).show();
      }

      $(document).ready(function() {
          clear_all_pages();
          $("#overview").show();
          $(".goto").click(function() {
              var target = $(this).attr("href");
              target = target.substring(1,target.length);
              change_page_to(target);
              return false;
          });
      });
    </script>
  </head>

  <body data-spy="scroll" data-target="logo-sidebar">

    <nav class="navbar navbar-default navbar-fixed-top" role="navigation">
      <div class="container-fluid">
        <div class="navbar-header">
          <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#bs-navbar-collapse-1" aria-expanded="false">
            <span class="sr-only">Toggle navigation</span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
            <span class="icon-bar"></span>
          </button>
          <a class="navbar-brand" href="#" onclick='clear_nav_bar_select(); clear_all_pages(); $("#overview").show(); return false;'>TACO</a>
        </div>

        <div class="collapse navbar-collapse" id="bs-navbar-collapse-1">
          <ul class="nav navbar-nav">
            <li><a href="#about" class="goto" id="about_button">About</a></li>
            <li><a href="#gettingstarted" class="goto" id="getting_started_button">Getting Started</a></li>
            <li><a href="#code" class="goto" id="code_button">Code</a></li>
            <li><a href="#faq" class="goto" id="faq_button">FAQ</a></li>
            <li><a href="#manual" class="goto" id="manual_button">Manual</a></li>
            <li><a href="#license" class="goto" id="license_button">License</a></li>
          </ul>
        </div>
      </div>
    </nav>

    <!-- Title bar -->
    <div class="container" id="overview">
      <header class="jumbotron subhead">
        <img src="img/taco_logo.png" height="200" style="height:200px;" alt="TACO icon">
        <h2>TACO: Large scale transcriptome meta-assembly for RNA-Seq</h2>
        <p class="lead">
        Transcriptome assemblers reconstruct full-length transcripts from
        the short sequence fragments generated by RNA-Seq. Large consortia
        such as <a href="http://cancergenome.nih.gov">TCGA</a>,
        <a href="http://www.gtexportal.org">GTex</a>,
        <a href="https://www.encodeproject.org">ENCODE</a>, the
        Cancer Cell Line Encyclopedia (CCLE), and many others have
        performed RNA-Seq on thousands of human tissues and cell lines,
        providing an unparalleled resource for investigating
        transcriptional diversity and complexity. Towards this end, we
        present Transcriptome Assemblies Combined into One (TACO), an
        algorithm that reconstructs a consensus transcriptome from a large
        collection of individual assemblies. TACO employs change point
        detection to break apart complex loci and correctly delineate
        transcript start and end sites, and a dynamic programming approach
        to assemble transcripts from a network of splicing patterns.
        TACO vastly outperforms meta-assembly software tools such
        as <a href="http://cole-trapnell-lab.github.io/cufflinks/cuffmerge">Cuffmerge</a>
        and <a href="https://ccb.jhu.edu/software/stringtie">Stringtie merge</a>.
        See the results of our comparison analysis <a href="#about">here</a>.
        </p>
      </header>
    </div>

    <div class="container">
      <!-- About Section -->
      <section id="about">
        <h2 class="page-header">Methods</h2>
        <div class="container">
          <div class="row">
            <div class="col-xs-8">
              <img class="img-responsive center-block" src="img/figure_1_schematic.png" width="100%" alt="Methods">
            </div>
            <div class="col-xs-4">
              <p class="lead">Transcriptome meta-assembly protocol</p>
              <ol>
                <li>Sequence reads from each dataset are aligned to the
                reference genome using
                <a href="https://github.com/alexdobin/STAR">STAR</a>,
                <a href="http://ccb.jhu.edu/software/hisat2">HISAT2</a>,
                or another RNA-Seq aligner</li>
                <li>Read alignments (SAM/BAM/CRAM format) from each dataset
                are assembled into transfrags using
                <a href="http://cole-trapnell-lab.github.io/cufflinks">Cufflinks</a>,
                <a href="https://ccb.jhu.edu/software/stringtie">StringTie</a>,
                or another <em>ab initio</em> transcriptome assembly tool.</li>
                <li>Meta-assembly is performed on the collection of
                <em>ab initio</em> assemblies (GTF format) by
                <strong>TACO</strong>, resulting in a consensus merged
                transcriptome</li>
                <li>Downstream analysis of gene expression, conservation,
                variation, and functionality can be performed</li>
              </ol>
            </div>
          </div>
        </div>

        <h2 class="page-header">Results</h2>
        <div class="container">
          <div class="row">
            <div class="col-xs-8">
              <img class="img-responsive center-block" src="img/figure_2a.png" width="100%" alt="Methods">
            </div>
            <div class="col-xs-4">
              <p class="lead">Varying the number of input assemblies</p>
              Performance metrics for TACO, Cuffmerge, and
              StringTie when merging different numbers of input assemblies. Recall
              (i.e., sensitivity), precision and the F-measure for all three
              tools were assessed for splicing patterns, splice junctions,
              and bases. Twenty batches were randomly selected at each
              number of assemblies. Points represent the mean statistic
              across the 20 runs, error bars represent the 95% confidence
              interval.
            </div>
          </div>

          <br/><br/>

          <div class="row">
            <div class="col-xs-8">
              <img class="img-responsive center-block" src="img/figure_2b.png" width="100%" alt="Methods">
            </div>
            <div class="col-xs-4">
              <p class="lead">Varying isoform fraction cutoff</p>
              <p>The parameter <code>--isoform-frac FRAC</code> configures
              TACO to only report isoforms with abundance ratio more than
              <code>FRAC</code> of the most abundant transcript in the locus</p>
              <p>The precision-recall (PR) plots depict the performance for the
              three merging tools at 50 different isoform fraction cutoffs ranging
              from 0.001-0.999. Statistics describe the merger of the 55 breast
              cancer cell lines in the CCLE.</p>
            </div>
          </div>

          <br/><br/>

          <div class="row">
            <div class="col-xs-8">
              <img class="img-responsive center-block" src="img/figure_2c.png" width="100%" alt="Results">
            </div>
            <div class="col-xs-4">
              <p class="lead">Performance for highly expressed transcripts</p>
              <p>The precision-recall (PR) plots depict the performance for the
              three merging tools across subsets of the top 500, 1k, 1.5k,
              2k, 3k, 5k, 7k, 10k, 12k, 15k, 18k, 20k, 22k, 25k, 27k, and 30k
              highest expressed transcripts in the merger of 55 breast cancer
              cell lines from the CCLE</p>
            </div>
          </div>

          <br/><br/>

          <div class="row">
            <div class="col-xs-8">
              <img class="img-responsive center-block" src="img/figure_2d.png" alt="Results">
            </div>
            <div class="col-xs-4">
              <p class="lead">Fraction of read-through transcripts</p>
              <p>Scatterplot depicting the fraction of read-through
              genes produced in the meta-assembly from the three merge
              tools. Gene designated as being a read-through if it contains
              at least one transcript isoform bearing exonic overlap with
              two separate protein-coding genes on the same strand. Fraction
              of read-throughs was determined for assemblies at 50 different
              isoform fraction cutoffs ranging from 0.001-0.999.</p>
            </div>
          </div>
        </div>
      </section>

      <!-- Getting Started Section-->
      <section id="gettingstarted">
        <h2 class="page-header">Getting Started</h2>
        <ul>
          <li><a href="#gettingstarted-install">Install</a></li>
          <li><a href="#gettingstarted-run">Run</a></li>
        </ul>

        <h3 id="gettingstarted-install">Installation</h3>

        <h4>Download Software</h4>
        <p>
          Download the latest TACO release
          <a href="#code" class="goto" id="code_button">here</a>
        </p>

        <h4>Install dependencies</h4>
          <p>TACO requires several dependencies:</p>
          <ul>
            <li><a href="http://www.python.org">Python 2.7</a></li>
            <li><a href="http://www.numpy.org">Numpy</a></li>
            <li><a href="http://cython.org">Cython</a> (required to compile C extension modules)</li>
          </ul>
          <p>
            We developed and tested TACO on Mac OSX and Linux using the
            <a href="https://www.enthought.com/products/canopy">Enthought Canopy</a> Python Distribution.
            If you do not already have a Python distribution installed on your system, we recommend this.
          </p>
        </p>

        <h4>Build</h4>
        <p>
          Unpack the release file (.tar.gz or .zip) into a local direectory <code>SOURCE_DIR</code>. For example:
          <pre>$ tar -zxvf taco-X.Y.Z.tar.gz</pre>
        </p>
        <p>
          Build and install the software into directory INSTALL_DIR
          <pre>$ cd SOURCE_DIR
$ python setup.py install --prefix INSTALL_DIR</pre>
        </p>

        <h4>Setup your Python environment</h4>
        <p>The TACO program will be installed to <code>INSTALL_DIR/bin/taco_run.py</code>.
        Associate modules will be available at <code>INSTALL_DIR/lib/python2.7/site-packages</code>
        </p>
        <p>If you installed TACO into a non-standard directory, your
        PYTHONPATH environment variable must include the TACO package
        directory. Using bash shell:</p>
        <pre>$ export PYTHONPATH=INSTALL_DIR/lib/python2.7/site-packages:$PYTHONPATH</pre>
        </p>

        <h3 id="gettingstarted-run">Running TACO</h3>
        <p>Run TACO from the command line as follows:</p>
        <pre>$ INSTALL_DIR/bin/taco_run.py &lt;options&gt; &lt;gtf_files.txt&gt;</pre>

        <p>A detailed description of the parameters used to control TACO is below:</p>

        <h4>Arguments</h4>

        <p><strong><code>&lt;gtf_files.txt&gt;</code></strong></p>
        <p>This required input to TACO is a text file containing paths to files in the
          <a href="http://mblab.wustl.edu/GTF22.html">GTF 2.2 format</a>,
          one per line, produced by an <em>ab initio</em> assembler.
          For example, the text file passed to TACO could look like this:</p>
          <pre>/path/to/my/assemblies/mcf7.gtf
/path/to/my/assemblies/bt474.gtf
/path/to/my/assemblies/hela.gtf</pre>
        </p>

        <h4>General Options</h4>

        <p><strong><code>-h/--help</code></strong></p>
        <p>Prints the help message and exits</p>

        <p><strong><code>-v/--verbose</code></strong></p>
        <p>Enables detailed logging for debugging purposes</p>

        <p><strong><code>-p/--num-processes N</code></strong></p>
        <p>Run TACO in parallel mode with N processes [default=1]</p>

        <p><strong><code>-o/--output-dir DIR</code></strong></p>
        <p>Directory where output files will be stored (must not already exist)</p>

        <p><strong><code>--gtf-expr-attr ATTR</code></strong></p>
        <p>GTF attribute field containing expression estimate [default=FPKM]</p>

        <p><strong><code>--min-transfrag-length N</code></strong></p>
        <p>Filter input transfrags with <code>length < N</code> prior to assembly [default=200]</p>

        <p><strong><code>--min-expr X</code></strong></p>
        <p>Filter input transfrags with <code>expression < X</code> (<code>gtf-expr-attr</code> parameter)
        prior to assembly [default=0.5]</p>

        <p><strong><code>--isoform-frac FRAC</code></strong></p>
        <p>Report transcript isoforms with expression fraction >=FRAC relative
        to the highest expressed gene [default=0.05]</p>

        <h4>Advanced Options</h4>
        <p>TACO has additional advanced options that can be used to
          configure its behavior. We recommend that users leave these
          options at their default settings.
        </p>

      </section>

      <section id="code">
        <h2 class="page-header">Download Code</h2>
        <h4>Releases</h4>
        <table class="table">
          <thead>
              <tr>
                  <th>Version</th>
                  <th>Date</th>
                  <th>Changes</th>
                  <th>Link</th>
              </tr>
          </thead>
          <tr>
              <td>v0.4.3 (most recent)</td>
              <td>3/24/2016</td>
              <td>Version used for manuscript submission</td>
              <td>
                Download
                <a href="https://github.com/yniknafs/taco/archive/v0.4.3.tar.gz">.tar.gz</a>
                <a href="https://github.com/yniknafs/taco/archive/v0.4.3.zip">.zip</a>
              </td>
          </tr>
        </table>

        <h4>Source Code</h4>
        <p>The TACO GitHub repository is located
        <a href="https://github.com/yniknafs/taco">here</a>.
        </p>
      </section>

            <section id="faq">
                <h2 class="page-header">FAQ</h2>
                <ul>
                    <li><b>Q. Sed ac ante augue. Maecenas felis tortor, venenatis ut venenatis ac, vehicula eu eros. Nam sit amet dolor massa.</b></li>
                    <li>A.  Integer consectetur, nunc id volutpat ultrices</li>
                    <br>
                    <li><b>Q. Nunc congue lacus ornare tempor aliquet. Nam venenatis lorem id quam iaculis mollis. Mauris faucibus vestibulum nisl in semper. Cras semper justo nec enim malesuada ornare. Suspendisse potenti. Etiam vitae feugiat mi. Quisque et consequat turpis.</b></li>
                    <li>A. Yes</li>
                    <br>
                    <li><b>Q. Vivamus dictum sodales ante in pellentesque. Vivamus elit mauris, eleifend at est laoreet, sollicitudin porta risus.</b></li>
                    <li>A. Nam eget sodales lorem, nec scelerisque sem.</li>
                    <br>
                    <li><b>Q. Fusce justo elit, pharetra vitae lorem et, maximus vestibulum mi.</b></li>
                    <li>A. Aliquam erat volutpat. Sed mauris mauris, consequat sed urna eget, gravida consectetur tortor. Cras euismod mauris et ante tempor sollicitudin. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Aliquam viverra lectus in sem egestas, a mollis mi pellentesque. In nulla enim, gravida at accumsan eget, efficitur nec ante. Nullam commodo risus vel faucibus pulvinar. Etiam tincidunt efficitur feugiat.</li>
                    <br>
                    <li><b>Q. Sed non ligula venenatis, mattis ex nec, bibendum augue.</b></li>
                    <li>A. Donec vel neque a elit euismod scelerisque. Quisque eget mauris rhoncus, elementum mi quis, vestibulum mauris. Proin bibendum sapien in lectus maximus, et pellentesque velit egestas.</li>
                </ul>
            </section>


            <section id="manual">
                <h2 class="page-header">Manual</h2>

                A full API is available at: <a href="https://readthedocs.org/projects/tacoman/">TACO - readthedocs</a>
            </section>

            <section id="license">
                <h2 class="page-header">License</h2>
                <h4>MIT License</h4>

                <pre>Copyright (c) 2015-2016 Matthew Iyer, Yashar Niknafs, Balaji Pandian, Chinnaiyan Research Lab -- University of Michigan

Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:

The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
                </pre>

            </section>

        </div>


    </body>


</html>
